File: lang/c/test/vac.fun/data_structures/LinkedQueueTest.c
#include <vac.fun/data_structures/LinkedQueue.h>
#include <vac.fun/util/TestRunner.h>

bool testLinkedQueue(void) {
    size_t size = 8192;

    LinkedQueue *queue = LinkedQueue_new();

    for (size_t i = 1; i <= size; i++) {
        LinkedQueue_enqueue(queue, (int) i);
    }

    if (LinkedQueue_size(queue) != size) {
        return false;
    }

    for (size_t i = 1; i <= size; i++) {
        if ((size_t) LinkedQueue_peek(queue) != i) {
            return false;
        }
        LinkedQueue_dequeue(queue);
    }

    return LinkedQueue_isEmpty(queue);
}

int main(void) {
    TestRunner_parseTest(testLinkedQueue());
}

File: lang/nodejs/test/vac.fun/data_structures/LinkedQueueTest.js
"use strict"

const LinkedQueue = require("vac.fun/data_structures/LinkedQueue")
const TestRunner  = require("vac.fun/util/TestRunner")

const testLinkedQueue = () => {
    let size = 8192

    const queue = new LinkedQueue.LinkedQueue()

    for (let i = 1; i <= size; i++) {
        queue.enqueue(i)
    }

    if (queue.size() !== size) {
        return false
    }

    for (let i = 1; i <= size; i++) {
        if (queue.peek() !== i) {
            return false
        }
        queue.dequeue()
    }

    return queue.isEmpty()
}

if (module === require.main) {
    TestRunner.parseTest(testLinkedQueue())
}
